lnwallet/wallet.go

```diff
@@ -510,7 +530,16 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 
 		localFundingAmt = fundingIntent.LocalFundingAmt()
 		remoteFundingAmt = fundingIntent.RemoteFundingAmt()
+
+	} else if shimIntent, ok := l.fundingIntents[req.PendingChanID]; ok {
+		// Alternatively, if we've just received an inbound funding
+		// request that we have a registered him intent to, then we'll
+		// obtain the backing intent now. In this case, we're doing a
+		// special funding workflow that allows more advanced
+		// constructions such as channel factories to be instantiated.
+		fundingIntent = shimIntent
 	}
+	// TODO(roasbeef): need mutex above
```

Comment:
reminder to use mutex, or pass intent via a `handleRegisterFundingIntent` to avoid mutex?

Reply:
To grab mutex, TODO resolved.